ml_metadata/tools/mlmd_bench/fill_types_workload_test.cc

```diff
@@ -23,59 +25,431 @@ limitations under the License.
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  for (int64 i = 0; i < num_execution_types_to_insert; i++) {
+    ExecutionType* curr_type = put_request.add_execution_types();
+    curr_type->set_name(absl::StrCat("pre_insert_execution_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  for (int64 i = 0; i < num_context_types_to_insert; i++) {
+    ContextType* curr_type = put_request.add_context_types();
+    curr_type->set_name(absl::StrCat("pre_insert_context_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  TF_RETURN_IF_ERROR(store->PutTypes(put_request, &put_response));
+  return tensorflow::Status::OK();
+}
+
+// Executes the given FillTypes workloads.
+tensorflow::Status ExecuteWorkload(MetadataStore* store,
+                                   std::unique_ptr<FillTypes>& workload) {
+  for (int64 i = 0; i < workload->num_operations(); ++i) {
+    OpStats op_stats;
+    TF_RETURN_IF_ERROR(workload->RunOp(i, store, op_stats));
+  }
+  return tensorflow::Status::OK();
+}
+
+// Checks types update status. If the updates are working properly, the type id
+// should remain the same after the updates. On the other hand, the properties
+// size for each type should be greater than before since some new fields have
+// been added in the update process.
+template <typename T>
+bool CheckTypesUpdateStatus(const Type& type_before, const Type& type_after) {
+  if (absl::get<T>(type_before).id() != absl::get<T>(type_after).id()) {
+    return false;
+  }
+  if (absl::get<T>(type_before).properties().size() >=
+      absl::get<T>(type_after).properties().size()) {
+    return false;
+  }
+  return true;
+}
 
 // Test fixture that uses the same data configuration for multiple following
-// tests.
-class FillTypesTest : public ::testing::Test {
+// parameterized FillTypes insert tests.
+// The parameter here is the specific Workload configuration that contains
+// the FillTypes insert configuration and the number of operations.
+class FillTypesInsertParameterizedTestFixture
+    : public ::testing::TestWithParam<WorkloadConfig> {
  protected:
   void SetUp() override {
     ConnectionConfig mlmd_config;
     // Uses a fake in-memory SQLite database for testing.
     mlmd_config.mutable_fake_database();
     TF_ASSERT_OK(CreateMetadataStore(mlmd_config, &store_));
-    FillTypesConfig fill_types_config;
-    fill_types_config = testing::ParseTextProtoOrDie<FillTypesConfig>(
-        R"(
-          update: false
-          specification: CONTEXT_TYPE
-          num_properties { minimum: 1 maximum: 10 }
-        )");
-    fill_types_ = absl::make_unique<FillTypes>(
-        FillTypes(fill_types_config, kNumOperations));
+    fill_types_insert_ = absl::make_unique<FillTypes>(
+        FillTypes(GetParam().fill_types_config(), GetParam().num_operations()));
   }
 
-  std::unique_ptr<FillTypes> fill_types_;
+  std::unique_ptr<FillTypes> fill_types_insert_;
```

Comment:
nit, fill_types_ looks ok as the class it insert fixture.

Reply:
Noted! Fixed!